docs/HowToArchiveJournal.md

```diff
@@ -17,13 +17,134 @@ For more info on NuoDB database journaling, please consult the official [NuoDB d
+```
+
+NOTE: NuoDB also requires the NuoDB admin layer, which is also a StatefulSet but won't be directly involved in this migration process.
+
+There will be a number of PersistentVolumeClaims.
+- 1 PVC for the NoHotCopy Storage Manager
+- 2 PVCs for the HotCopy Storage Manager
+- 1 PVC for the admin layer
+
+```shell
+k get persistentvolumeclaims -n nuodb
+NAME                                                       STATUS   VOLUME                                     CAPACITY   ACCESS MODES
+archive-volume-sm-database-nuodb-cluster0-demo-0           Bound    pvc-e8227428-6b9f-47b1-acc0-41b70a21d043   20Gi       RWO
+archive-volume-sm-database-nuodb-cluster0-demo-hotcopy-0   Bound    pvc-19540f97-8407-4fe1-a940-3b3f83538f0f   20Gi       RWO
+backup-volume-sm-database-nuodb-cluster0-demo-hotcopy-0    Bound    pvc-daebd030-a4eb-46e1-8451-6d65e4fa3061   20Gi       RWO
+raftlog-admin-nuodb-cluster0-0                             Bound    pvc-4930b80b-a622-487a-90a6-60a2bd3f0548   1Gi        RWO
+
+```
+
+
+#### Upgrade the NoHotCopy StatefulSet
+First, scale the StatefulSet to 0.
+```shell
+kubectl scale statefulset sm-database-nuodb-cluster0-demo --replicas=0
+statefulset.apps/sm-database-nuodb-cluster0-demo scaled
+```
+
+Second, delete all PVCs
+```shell
+kubectl delete pvc -n nuodb <PVC_NAME>
+```
+
+In this case, the `<PVC_NAME>` will be `archive-volume-sm-database-nuodb-cluster0-demo-0`.
+
+Third, delete the StatefulSet:
+```shell
+kubectl delete statefulset sm-database-nuodb-cluster0-demo
+```
+
+Fourth, reinstall the StatefulSet using Helm.
+To enable the journal on the deleted StatefulSet, use the following value:
+```
+database.sm.noHotCopy.journalPath.enabled=true
+```
+
+```shell
+helm install database nuodb/database \
+--set database.sm.noHotCopy.journalPath.enabled=true \
+--set database.sm.noHotCopy.replicas=1
+```
+
+Wait for the Storage Manager Pod to become READY before proceeding.
+
+#### Upgrade the HotCopy StatefulSet
+First, scale the StatefulSet to 0.
+```shell
+kubectl scale statefulset sm-database-nuodb-cluster0-demo-hotcopy --replicas=0
+statefulset.apps/sm-database-nuodb-cluster0-demo-hotcopy scaled
+```
+
+Second, delete all PVCs
+```shell
+kubectl delete pvc -n nuodb <PVC_NAME>
+```
+
+In this case, the `<PVC_NAME>` will be `archive-volume-sm-database-nuodb-cluster0-demo-hotcopy-0`.
+
+Third, delete the StatefulSet:
+```shell
+kubectl delete statefulset sm-database-nuodb-cluster0-demo-hotcopy
+```
+
+Fourth, reinstall the StatefulSet using Helm.
+To enable the journal on the deleted StatefulSet, use the following value:
+```
+database.sm.HotCopy.journalPath.enabled=true
+```
+
+```shell
+helm install database nuodb/database \
+--set database.sm.noHotCopy.journalPath.enabled=true \
+--set database.sm.HotCopy.journalPath.enabled=true \
+--set database.sm.noHotCopy.replicas=1
+```
+
+Wait for the Storage Manager Pod to become READY.
+
+Clean up the remaining `backup-` PersistentVolumeClaims.
```

Comment:
Why a cleanup is needed? Does this mean that the user will lose all backups by provisioning another PV for the backup PVC?

Reply:
I dont think that they will get remounted to the new SS. I will test this before I speak further

Reply:
They should do - but we should specify that the same release name is used to ensure that is the case

Reply:
I removed the cleanup step. It can be inherited